### PR TITLE
Fix registry host not being synced with its secret

### LINKED ISSUE #10314: Registry authentication method is modified unexpectedly
Internal reference: SURE-7465
Reported in 2.8.0

Issue description:
Registry authentication method is modified unexpectedly.

Actual behavior:
Authentication of docker.io Registry is modified from None to HTTP Basic Auth.

Expected behavior:
Authentication of docker.io Registry is not modified.

No workaround.

Repro steps:

1. [Create a Downstream Cluster]
- Create -> Amazon EC2 (RKE2)
- v1.27.8+rke2r1
- CNI Calico

2. [Configure Registry]
- Container Registry: docker.io
- Authentication: None

3. Mirrors
- Registry Hostname: $REGISTRY_HOSTNAME

Registry Authentication
Registry Hostname: $REGISTRY_HOSTNAME
Authentication: Create a HTTP Basic Auth Secret
Username: user
Password: pass

4. After Cluster created,
- Cluster Management -> Edit Config -> Registries

Authentication of docker.io Registry is modified from None to HTTP Basic Auth.

### PR DESCRIPTION
**Pull request by momesgin**
### Summary
Fixes #10314 & #10093 

There are more issues related to the Registries section(like #10371) that might affect this, but for now this change should only fix #10314 without breaking other functionalities there.

`registrySecret`(shown as Authentication field under Container Host field) was being extracted from the first host name in list of host names in `registries.configs`, but this field should exactly look for the container host name(`registryHost`) in that list.

### Screenshot/Video

https://github.com/rancher/dashboard/assets/135728925/143f785d-d174-4a4e-bae9-76524499d177

### CODE REVIEW

**Review by richard-cox — CHANGES_REQUESTED** [review]
#10314 is resolved, but #10093 is still reproducible.

the fix for #10314 is good, i gave a go with a system registry config as well... and that works fine

File: cypress/e2e/tests/pages/manager/registries-rke2.spec.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
+import ClusterManagerListPagePo from '@/cypress/e2e/po/pages/cluster-manager/cluster-manager-list.po';
+import ClusterManagerCreateRke2CustomPagePo from '@/cypress/e2e/po/edit/provisioning.cattle.io.cluster/create/cluster-create-rke2-custom.po';
+import { machineSelectorConfigPayload, registriesWithSecretPayload } from '@/cypress/e2e/blueprints/manager/registries-rke2-payload';
+
+const registryHost = 'docker.io';
+const registryAuthHost = 'a.registry.com';
+const clusterName = `test-cluster-${ Math.random().toString(36).substr(2, 6) }`;
```
[the remark this answers is not in the document]
Comment (momesgin) [reply cypress/e2e/tests/pages/manager/registries-rke2.spec.ts]: Thanks for letting me know about this, done.

File: shell/edit/provisioning.cattle.io.cluster/tabs/registries/RegistryConfigs.vue [hunk 2]
```diff
@@ -145,6 +145,7 @@ export default {
               v-model="row.value.hostname"
               label="Registry Hostname"
               :mode="mode"
+              data-testid="registry-auth-host-input"
```
[the remark this answers is not in the document]
Comment (momesgin) [reply shell/edit/provisioning.cattle.io.cluster/tabs/registries/RegistryConfigs.vue]: thanks, done.

File: shell/edit/provisioning.cattle.io.cluster/rke2.vue [hunk 3]
```diff
@@ -1568,8 +1568,7 @@ export default {
         set(regs, 'mirrors', {});
       }
 
-      const hostname = Object.keys(regs.configs)[0];
-      const config = regs.configs[hostname];
+      const config = regs.configs[this.registryHost];
```
[the remark this answers is not in the document]
Comment (momesgin) [reply shell/edit/provisioning.cattle.io.cluster/rke2.vue:1576]: @richard-cox thanks for spending the time on this and sharing your findings 🙏, I was confused too about the distinction between the root url and the rest of the registry urls.

Regarding #10093 maybe I'm missing something, I saw your comment pointing to the condition in `rke2.vue`:
``` 
    if ( hostname && this.registrySecret ) {
        ...
```
are you suggesting that the root url should be added to `registries.configs` no matter if it has a secret or not? my thinking was having it set as `system-default-registry` is sufficient. I've recorded a video trying different cases for the root url as well as the other registry url without selecting a secret. 

Notes about the video:
- I've added a console log right before the condition^
- ignore the `create-yaml.js` errors on console, it's not affecting these scenarios. There's an [open PR](https://github.com/rancher/dashboard/pull/10422) for it too.
- if you try adding CA Cert Bundle you get an error(at the end of the video) that I think it's unrelated too


https://github.com/rancher/dashboard/assets/135728925/8da0146e-a2d0-4bf0-8ec8-2ba06d89c125

File: cypress/e2e/po/components/registries-rke2.po.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  constructor(selector = '.dashboard-root') {
+    super(selector);
+  }
+
+  enableRegistryCheckbox() {
+    return new CheckboxInputPo('[data-testid="registries-enable-checkbox"]');
+  }
+
+  clickTab(selector: string) {
+    new TabbedPo().clickTabWithSelector(selector);
```
[the remark this answers is not in the document]
Comment (momesgin) [reply cypress/e2e/po/components/registries-rke2.po.ts]: done

File: cypress/e2e/po/components/registries-rke2.po.ts [hunk 5]
```diff
@@ -0,0 +1,49 @@
+import ComponentPo from '@/cypress/e2e/po/components/component.po';
+import TabbedPo from '@/cypress/e2e/po/components/tabbed.po';
+import CheckboxInputPo from '@/cypress/e2e/po/components/checkbox-input.po';
+import LabeledInputPo from '@/cypress/e2e/po/components/labeled-input.po';
+import SelectOrCreateAuthPo from '@/cypress/e2e/po/components/select-or-create-auth.po';
+
+export default class RegistriesRke2 extends ComponentPo {
```
[the remark this answers is not in the document]
Comment (momesgin) [reply cypress/e2e/po/components/registries-rke2.po.ts:6]: Sorry I'm not quiet sure what you suggest here, do we have an example in our codebase?

[the remark this answers is not in the document]
Comment (richard-cox) [reply cypress/e2e/po/components/registries-rke2.po.ts:6]: If you check out the PO that will map to Components
cypress/e2e/po/components/unit-input.po.ts --> shell/components/form/UnitInput.vue 
cypress/e2e/po/detail/management.cattle.io.user.po.ts --> shell/detail/management.cattle.io.user.vue
cypress/e2e/po/pages/about.po.ts --> shell/pages/about.vue

For this case 
shell/edit/provisioning.cattle.io.cluster/tabs/registries/index.vue --> PO for the registry tab aka  RegistriesTabPo
shell/edit/provisioning.cattle.io.cluster/tabs/registries/RegistryConfigs.vue --> PO for the registry config aka RegistyConfigsPo

as tabs/registries/index.vue contains the registryconfig component the PO hierarchy would be similar. so something like
```
RegistriesTabPo {
  registryConfigs(): RegistyConfigsPo {
   new RegistyConfigsPo()
 }
}```

File: shell/edit/provisioning.cattle.io.cluster/rke2.vue (hunk 3 above)
[the remark this answers is not in the document]
Comment (richard-cox) [reply shell/edit/provisioning.cattle.io.cluster/rke2.vue:1576]: Not sure what happened, but i could reproduce https://github.com/rancher/dashboard/issues/10093 in this PR. I've also checked head and can no longer reproduce it there either. 

Looked through the code again and updated my comment. The root registry should go in the machineSelectorConfig config object, the first without a selector. I've confirmed that's happening in this PR

**Review by richard-cox — CHANGES_REQUESTED** [review]
just 1 e2e change to go

File: cypress/e2e/po/components/registries-rke2.po.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (momesgin) [reply cypress/e2e/po/components/registries-rke2.po.ts]: thanks for the details 🙏, done

**Review by richard-cox — APPROVED** [review]
Ship it!

**Pull request merged**